ext/liquid_c/vm.c

```diff
@@ -284,10 +341,37 @@ static VALUE vm_render_until_error(VALUE uncast_args)
                 vm_stack_push(vm, result);
                 break;
             }
-            case OP_PUSH_EVAL_EXPR:
+
+            // Rendering instructions
+
+            case OP_WRITE_RAW:
             {
-                VALUE expression = (VALUE)*const_ptr++;
-                vm_stack_push(vm, context_evaluate(args->context, expression));
+                const char *text = (const char *)*const_ptr++;
+                size_t size = *const_ptr++;
+                rb_str_cat(output, text, size);
+                increment_write_score(args);
+                break;
+            }
+            case OP_WRITE_NODE:
+                rb_funcall(cLiquidBlockBody, id_render_node, 3, args->context, output, (VALUE)*const_ptr++);
+                if (RARRAY_LEN(vm->interrupts)) {
+                    return false;
+                }
+                increment_write_score(args);
+                break;
+            case OP_RENDER_VARIABLE_RESCUE:
+                args->node_line_number = (unsigned int)*const_ptr++;
+                args->ip = ip;
```

Comment:
`ip` and `const_ptr` are saved because next instruction is assumed to be calling a Ruby function?

Reply:
This PR just moves this case to keep the rendering instruction handling together, since they shouldn't be used in the Liquid::C::Expression code.

This was added in the last PR (https://github.com/Shopify/liquid-c/pull/59) and described as follows

> Another design choice I made for this PR is to do error handling for the whole block body render, rather than for each variable render. This way we can reduce the state saving cost of `rb_rescue` from liquid code that doesn't encounter variable render errors. In order to recover from the exception, we just restore the stack size to what it was at the start of the block body render and iterate the instructions to jump just past the variable write.

In order to iterate the instructions to the end of the variable render on an exception, we need to save the instruction position at the start of variable rendering.

I'll add some comments to the code to make this clearer.

Reply:
I'll also move code around in the last PR so that the diff is easier to read in this PR.